### ISSUE TITLE
Google Oauth2 Stopped Working

### ISSUE THREAD

**Issue by tomhc** [yiisoft/yii2-authclient#245]
### What steps will reproduce the problem?

Login to Yii2 web site with Google Authentication

### What's expected?

Successful login

### What do you get instead?

```
Exception (Exception) 'yii\authclient\InvalidResponseException' with message 'Request failed with code: 400, message: {
  "error": "redirect_uri_mismatch",
  "error_description": "Bad Request"
}' 

in backend_v1/vendor/yiisoft/yii2-authclient/src/BaseOAuth.php:229

Stack trace:
#0 backend_v1/vendor/yiisoft/yii2-authclient/src/OAuth2.php(124): yii\authclient\BaseOAuth->sendRequest(Object(yii\httpclient\Request))
#1 backend_v1/vendor/yiisoft/yii2-authclient/src/AuthAction.php(392): yii\authclient\OAuth2->fetchAccessToken('4/WQAQbvB9Okb-S...')
#2 backend_v1/vendor/yiisoft/yii2-authclient/src/AuthAction.php(209): yii\authclient\AuthAction->authOAuth2(Object(dektrium\user\clients\Google))
#3 backend_v1/vendor/yiisoft/yii2-authclient/src/AuthAction.php(194): yii\authclient\AuthAction->auth(Object(dektrium\user\clients\Google))
#4 [internal function]: yii\authclient\AuthAction->run()
#5 backend_v1/vendor/yiisoft/yii2/base/Action.php(94): call_user_func_array(Array, Array)
#6 backend_v1/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\Action->runWithParams(Array)
#7 backend_v1/vendor/yiisoft/yii2/base/Module.php(528): yii\base\Controller->runAction('auth', Array)
#8 backend_v1/vendor/yiisoft/yii2/web/Application.php(103): yii\base\Module->runAction('user/security/a...', Array)
#9 backend_v1/vendor/yiisoft/yii2/base/Application.php(386): yii\web\Application->handleRequest(Object(yii\web\Request))
#10 backend_v1/web/index.php(14): yii\base\Application->run()
#11 {main}
```

### Additional info

| Q                       | A
| ----------------------- | ---
| Yii version             | 2.0.14
| Yii Auth Client version |  2.1.6 and 2.1.5
| Yii HTTP Client version | 2.0.6
| PHP version             | 7.2.5
| Operating system        | Ubuntu LTS 16.04

- Google auth broke across all of our web sites yesterday (Sep 12), about a dozen different domains. 
- An angular site Google auth also does not work
- Google accounts can login to other Google Oauth web sites
- A brand new Google API project, client, etc. was created, under a different Google Developer console account, same result.
- Sites with code that has not changed in weeks also broke. It's not a bug in our code.

It looks like something changed with the Google API, but that seems very unlikely. For further diagnosis, has anyone else seen this happen?

**Comment by samdark** [comment]
Have you updated code?

**Comment by samdark** [comment]
Duplicate of #241

**Comment by Juanfcomoraless** [comment]
When will be solved this bug?

**Comment by samdark** [comment]
It is already in master.

**Comment by Juanfcomoraless** [comment]
> It is already in master.
Thanks, Its apply to nodge library.

**Comment by HeribertoQL** [comment]
does this apply to version 2.0.5 of yii?

**Comment by tomhc** [comment]
Same result with a different base domain and Google Developer account, so the problem isn't due to either of those factors.

**Comment by spinitron** [comment]
It seems Google changed something that caused yii\authclient\OAuth2::defaultReturnUrl to include the scope query parameter which interferes with the pattern matching Google uses to validate return URLs against the whitelist you set up.

**Comment by tomhc** [comment]
There is a second version of the API, https://accounts.google.com/o/oauth2/v2/auth. It's well documented. No sign of the v1 API in the docs, and it may no longer be supported. Our non-Yii apps broke as well, and it seems best to upgrade them to v2. Thank you to everyone for your quick responses.

**Comment by newscloud** [comment]
I just composer updated to latest 2.0.15.1 and "yiisoft/yii2-authclient" ... the bug still happens on my local test system.

**Comment by newscloud** [comment]
This is now fixed with Yii 2.0.15.1 hotfix and Auth
yiisoft/yii2-authclient  2.1.7

Thanks Alex!

**Comment by softappetite** [comment]
Hi,
its work for facebook but when i try with google its not working
i got this following error:
![Screenshot_2021-03-17 Error – APIs Services – hotelandresort – Google Cloud Platform](https://user-images.githubusercontent.com/8993528/111443457-96527400-8733-11eb-9d2d-d3208d11666d.png)

Exception – yii\authclient\InvalidResponseException
Request failed with code: 403, message: {
"error": {
"code": 403,
"message": "Legacy People API has not been used in project 60300072034 before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/legacypeople.googleapis.com/overview?project=60300072034 then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry.",
"status": "PERMISSION_DENIED",
"details": [
{
"@type": "type.googleapis.com/google.rpc.Help",
"links": [
{
"description": "Google developers console API activation",
"url": "https://console.developers.google.com/apis/api/legacypeople.googleapis.com/overview?project=60300072034"
}
